### PR TITLE
Add config option which requires authentication in FreeURL mode

### PR DESCRIPTION
**Pull request by nidico**
### Component/Part
Configuration / backend.

### Description
This PR mitigates the issue described in #754 by introducing a config option which - if enabled - only allows authenticated users to create new FreeURL notes.

This is implemented in a backward compatible way, i.e. if the default option is applied, nothing changes.

### Steps

- [x] Added implementation
- [ ] Added / updated tests
- [x] Added / updated documentation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  signed-off my commits to accept the DCO.

### Related Issue(s)

#754 -  Unintended note creation in FreeURL mode

### CODE REVIEW

File: lib/config/environment.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ module.exports = {
   allowAnonymous: toBooleanConfig(process.env.CMD_ALLOW_ANONYMOUS),
   allowAnonymousEdits: toBooleanConfig(process.env.CMD_ALLOW_ANONYMOUS_EDITS),
   allowFreeURL: toBooleanConfig(process.env.CMD_ALLOW_FREEURL),
+  requireFreeURLAuthentication: toBooleanConfig(process.env.CMD_REQUIRE_FREEURL_AUTH),
```
Comment (SISheogorath) [inline lib/config/environment.js:32]: Either the environment variable should be written out or the JS variable name should be cut. Let's try to keep the variable names in sync.

File: lib/config/hackmdEnvironment.js [hunk 2]
```diff
@@ -24,6 +24,7 @@ module.exports = {
   allowAnonymous: toBooleanConfig(process.env.HMD_ALLOW_ANONYMOUS),
   allowAnonymousEdits: toBooleanConfig(process.env.HMD_ALLOW_ANONYMOUS_EDITS),
   allowFreeURL: toBooleanConfig(process.env.HMD_ALLOW_FREEURL),
+  requireFreeURLAuthentication: toBooleanConfig(process.env.HMD_REQUIRE_FREEURL_AUTH),
```
Comment (SISheogorath) [inline lib/config/hackmdEnvironment.js:27]: This is not not needed. We don't add new config to a compatibly layer.

File: lib/config/environment.js (hunk 1 above)
Comment (nidico) [reply lib/config/environment.js]: Okay, I have decided for the long version as *auth* could also mean *authorization*.

File: lib/config/hackmdEnvironment.js (hunk 2 above)
Comment (nidico) [reply lib/config/hackmdEnvironment.js]: Okay, removed. I honestly didn't even know what this file was good for, but I have assumed it's there for legacy reasons...

**Comment by nidico** [comment]
(push-forced because of missing sign-off)

**Review by SISheogorath — APPROVED** [review]
No further concerns from my side

**Review by davidmehren — APPROVED** [review]

**Comment by davidmehren** [comment]
Thanks!

I hope to release 1.8 with this change in the next few days.

**Comment by nidico** [comment]
Thanks for reviewing and merging!

**Pull request merged**